Tests/UnitTests/Analytics/EventAnalyticsProviderTests.swift

```diff
@@ -0,0 +1,179 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import XCTest
+@_spi(AdyenInternal) @testable import Adyen
+@testable import AdyenNetworking
+
+final class EventAnalyticsProviderTests: XCTestCase {
+    
+    var eventDataSource = AnalyticsEventDataSource()
+
+    func testShouldNotSendEventsWhenNoAttemptId() {
+        let apiClient = APIClientMock()
+        let analyticsResponse = EmptyResponse()
+        let analyticsResult: Result<Response, Error> = .success(analyticsResponse)
+        
+        apiClient.mockedResults = [analyticsResult]
+        
+        let expectation = expectation(description: "should not be called")
+        expectation.isInverted = true
+        
+        apiClient.onExecute = { _ in
+            expectation.fulfill()
+        }
+        
+        let sut = createSUT(apiClient: apiClient)
+
+        sut.sendEventsIfNeeded()
+        
+        wait(for: [expectation], timeout: 0.1)
+    }
+
+    func testShouldNotSendEventWhenAttemptIdButNoEvents() {
+        let apiClient = APIClientMock()
+        let sut = createSUT(apiClient: apiClient)
+        
+        let expectation = expectation(description: "should not be called")
+        expectation.isInverted = true
+        
+        apiClient.onExecute = { request in
+            if request is AnalyticsRequest {
+                expectation.fulfill()
+            }
+        }
+        
+        sut.sendEventsIfNeeded()
+        
+        wait(for: [expectation], timeout: 0.1)
+    }
+    
+    func testEventShouldSendWhenAttemptIdAndEventsExist() {
+        let apiClient = APIClientMock()
+        let sut = createSUTWithSuccessMock(apiClient: apiClient)
+        
+        let infoEvent = AnalyticsEventInfo(component: "card", type: .rendered)
+        sut.add(info: infoEvent)
+        
+        let shouldSendExpectation = expectation(description: "send event is called")
+        let expectedId = infoEvent.id
```

Comment:
Is this `expectedCheckoutAttemptId` or a local event id?

Reply:
It's the id of that info event, no relation to the attempt id

Reply:
🤔 

Then I have to admit it's not clear when do you set a `checkoutAttemptId` itself as multiple test names suggest it should be either set of nil.

Reply:
Hmm, since attempt id is needed for all these tests, it's set on the helper function.
What part is confusing?

Reply:
I replied in the comment above.